sdks/python/apache_beam/examples/ml-orchestration/kfp/components/ingestion/component.yaml

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Ingestion
+description: Component that mimicks scraping data from the web and outputs it to a jsonlines format file
+inputs:
+  - name: base_artifact_path
+    description: base path to store data
+    type: String
+outputs:
+  - name: ingested_dataset_path
+    description: target uri for the ingested dataset
+    type: String
+implementation:
+  container:
+    image: <your-docker-registry/ingestion-image-name:latest>
```

Comment:
This should indeed be modified by the user. As far as I'm aware that's not possible for component definitions in `.yaml` files without additional tooling. 

We could switch to using an independent templating engine like [jinja](https://jinja.palletsprojects.com/en/3.1.x/). In that case, there will be jinja-specific syntax in both the `yaml` files and the `pipeline.py` file that loads the kfp components. 

I believe that would indeed make it easier to run the example itself, but might make the snippets in the website's documentation harder to understand since it introduces yet another tool. So I'm open to suggestion about how to tackle this trade-off!

Reply:
I would prioritize the code snippets in the website, so I think it's okay to leave it like this. Let's just make sure to document which files require source code modification.

I'm guessing that would mean we wouldn't be able to run them via `python -m apache_beam.examples...`, right? If so, would it require cloning the repo? That should all be documented.

Reply:
Done! All arguments that were global config variables are changed to command line arguments. I did not add a templating engine for the `.yaml` files, but I did document on the website that a user should modify these with the correct container image url before the kfp pipeline can be executed.

Reply:
Just out of curiosity, is it possible for users to create new files and point the pipeline to those rather than modifying the existing files? If it's not possible, that's fine. Thanks for addressing everything!

Reply:
Yes it's possible. But then the `pipeline.py` should be modified to load the component from the new `.yaml` file. So then the user would still have to modify a file.

Reply:
Could the yaml file path be a command line argument? I was thinking maybe we could instruct users to copy that "template" yaml file and point the pipeline to use their file. That would allow to run with `python -m apache_beam.examples...` since the actual package source code doesn't have to be modified, but it would be a user-supplied file. What are your thoughts on it?

Reply:
That would be an option as well indeed. I'm just a bit worried that it again adds a little more complexity to the docs themselves, of course in favor of having an example that is easier to run. If that's fine, we can add it.

Reply:
To keep documentation cleaner, and since we still need to modify a file, let's just keep it as it is. Users will have to clone the repo and modify these files in place.